src/ResourceManager/Profile/Commands.Profile.Test/LongRunningCmdletTests.cs

```diff
@@ -0,0 +1,226 @@
+
+        AzureStreamTestCmdlet SetupCmdlet(bool CallShouldProcess, bool CallShouldContinue, out Mock<ICommandRuntime> mockRuntime)
+        {
+            var cmdlet = new AzureStreamTestCmdlet();
+            cmdlet.Warning.Add(Warning);
+            cmdlet.Debug.Add(Debug);
+            cmdlet.Verbose.Add(Verbose);
+            cmdlet.Progress.Add(Progress);
+            cmdlet.Output.Add(Output);
+            cmdlet.Error.Add(Error);
+            mockRuntime = new Mock<ICommandRuntime>();
+            if (CallShouldProcess || CallShouldContinue)
+            {
+                cmdlet.CallShouldProcess = true;
+                mockRuntime.Setup(m => m.ShouldProcess(It.IsAny<string>())).Returns(true);
+                mockRuntime.Setup(m => m.ShouldProcess(It.IsAny<string>(), It.IsAny<string>())).Returns(true);
+                mockRuntime.Setup(m => m.ShouldProcess(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>())).Returns(true);
+            }
+            else
+            {
+                mockRuntime.Setup(m => m.ShouldProcess(It.IsAny<string>())).Throws(new InvalidOperationException("ShouldProcess should not be called"));
+                mockRuntime.Setup(m => m.ShouldProcess(It.IsAny<string>(), It.IsAny<string>())).Throws(new InvalidOperationException("ShouldProcess should not be called"));
+                mockRuntime.Setup(m => m.ShouldProcess(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>())).Throws(new InvalidOperationException("ShouldProcess should not be called"));
+            }
+
+            if (CallShouldContinue)
+            {
+                cmdlet.CallShouldContinue = true;
+                mockRuntime.Setup(m => m.ShouldContinue(It.IsAny<string>(), It.IsAny<string>())).Returns(true);
+            }
+            else
+            {
+                mockRuntime.Setup(m => m.ShouldContinue(It.IsAny<string>(), It.IsAny<string>())).Throws(new InvalidOperationException("ShouldContinue should not be called"));
+            }
+
+            cmdlet.CommandRuntime = mockRuntime.Object;
+
+            return cmdlet;
+        }
+
+        void ValidateCompletedCmdlet<T>(AzureLongRunningJob<T> job) where T : AzurePSCmdlet
+        {
+            Assert.Equal("Completed", job.StatusMessage);
+            Assert.True(job.HasMoreData);
+            Assert.Collection(job.Debug, t => Assert.Equal(Debug, t.Message));
+            Assert.Collection(job.Warning, t => Assert.Equal(Warning, t.Message));
+            Assert.Collection(job.Verbose, t => Assert.Equal(Verbose, t.Message));
+            Assert.Collection(job.Progress, t => Assert.Equal(Progress, t));
+            Assert.Collection(job.Output, t => Assert.Equal(Output, t.BaseObject));
+            Assert.Collection(job.Error, t => Assert.Equal(Error, t));
+        }
+
+        public class TestCmdletOutput
+        {
+            public string Property { get { return "PropertyValue"; } }
+        }
+
+        public class AzureStreamTestCmdlet : AzurePSCmdlet
```

Comment:
@markcowl is there a way for us to mock out a cmdlet that implements `AzureRmLongRunningCmdlet` so we can write tests to validate that providing the `AsJob` parameter will run the cmdlet as a job, and not providing the parameter will run the cmdlet normally?

Reply:
Yes, but there isn't a good way to run ProcessRecord in a mocked test.  I think the best way to cover this is to cover the extension method that dewtermines whether to run this way, and then a scenario test.

Meanwhile, the eitire set of scenario tests validates that cmdlets with AsJob operate normally.